src/main/java/bio/terra/janitor/service/cleanup/flight/GoogleBucketCleanupStep.java

```diff
@@ -21,11 +24,19 @@ public GoogleBucketCleanupStep(ClientConfig clientConfig, JanitorDao janitorDao)
   protected StepResult cleanUp(CloudResourceUid resourceUid) {
     // TODO(yonghao): Set bucket lifetime to 0 and wait once Stairway supports waiting. This is more
     // robust for buckets with many or very large objects.
-    // TODO(PF-29): Iterate through and delete all objects within the bucket.
     try {
-      storageCow.delete(resourceUid.getGoogleBucketUid().getBucketName());
+      System.out.println("~~~~~~COMPLETE START");
+      Stopwatch stopwatch = Stopwatch.createStarted();
+      // Delete all Blobs before deleting bucket.
+      String bucketName = resourceUid.getGoogleBucketUid().getBucketName();
+      BucketCow bucketCow = storageCow.get(bucketName);
+      bucketCow.list().iterateAll().forEach(BlobCow::delete);
```

Comment:
iterating over something while deleting them can have issues. Does this work? Should we instead collect all of the blob ids, then delete all of them? That won't work well if there are so many blobs that it OOMs us, but short of that, it seems a little safer. If we have that many blobs in a bucket, we may want to use the lifetime deletion then.
WDYT?